### PR TITLE
YLP-3306 Display UDI from a pdf stored on an external source

### PR DESCRIPTION
**Pull request by ydrisR**
Embed the pdf document in yourloops (doc is stored on a public location such as s3) instead of hardcoding the content in blip.

### CODE REVIEW

File: packages/yourloops/pages/product-labelling/product-labelling-page.tsx [hunk 1]
```diff
@@ -29,35 +29,24 @@ import React, { type FC, useState } from 'react'
...
-        return rawHtmlFR
-      case LanguageCodes.Nl:
-        return rawHtmlNL
-      case LanguageCodes.It:
-        return rawHtmlIT
-      case LanguageCodes.En:
-      default:
-        return rawHtmlEN
+    let locale = 'en'
+    if (Object.values(LanguageCodes).includes(getCurrentLang())) {
```
Comment (celia-rouis) [inline packages/yourloops/pages/product-labelling/product-labelling-page.tsx:42]: **🎯 suggestion (blocking):** ‏The method `getHtml()` is not useful anymore, we can directly put the `<object>` tag in the main component (and remove the `html` state and the `parse(html)`)

Comment (ydrisR) [reply packages/yourloops/pages/product-labelling/product-labelling-page.tsx:42]: 👍

File: packages/yourloops/pages/product-labelling/product-labelling-page.tsx [hunk 2]
```diff
@@ -69,7 +58,7 @@ export const ProductLabellingPage: FC = () => {
   setPageTitle(t('product-labelling'))
 
   return (
-    <Box marginBottom={2}>
+    <Box marginBottom={2} width="100%" height="100%">
```
[the remark this answers is not in the document]
Comment (ydrisR) [reply packages/yourloops/pages/product-labelling/product-labelling-page.tsx:61]: ok

File: packages/yourloops/pages/product-labelling/product-labelling-page.tsx [hunk 3]
```diff
@@ -29,37 +29,44 @@ import React, { type FC, useState } from 'react'
...
-      case LanguageCodes.It:
-        return rawHtmlIT
-      case LanguageCodes.En:
-      default:
-        return rawHtmlEN
-    }
+
+    return `
+        <object
+            role="document"
```
Comment (celia-rouis) [inline packages/yourloops/pages/product-labelling/product-labelling-page.tsx:62]: **🎯 suggestion (non-blocking):** ‏We don't need to have a separate method to handle the `object` tag, we can put it directly in the main `return`.

To do that, we need to:
- Remove the state based on the `html` and create a state based on the `filename` (it's the only thing that depends on the language)
```ts
const [filePath, setFilePath] = useState(getFilePath())
```
- Create a method to get the whole file path
```ts
const getFilePath = (): string => {
  return `${config.ASSETS_URL}${getFileName()}.pdf`
}
```
- Update the listener on the language to update the file path (and not the html) when the language changes
```ts
i18n.on('languageChanged', () => {
  setFilePath(getFilePath())
})
```
- And finally, remove the method `getHtml()` and put the `object` tag in the main `return()` of the page, and make it depend on the `filePath` state
```ts
<Box marginBottom={2} sx={{ width: '100%', height: '100%' }}>
  <object
    role="document"
    type="application/pdf"
    data={filePath}
    width="100%" height="100%"
  />
</Box>
```

**Review by celia-rouis — APPROVED** [review]

**Pull request merged**